talks/broad_12_mar_2018.md

```diff
@@ -0,0 +1,558 @@
+for a fast coalescent simulator, [msprime](https://github.com/jeromekelleher/msprime).
+
+- stores genealogical *and* variation data **very** compactly
+
+- efficient algorithms available:
+
+    * subsetting
+    * calculation of allele frequencies in arbitrary cohorts
+    * linkage disequilibrium
+    * log-time haplotype matching
+
+- tree-based sequence storage closely related to haplotype-matching compression
+
+
+## Simulated file sizes
+
+::: {.columns}
+::::::: {.column width="50%"}
+
+![file sizes](file_size.png)
+
+:::
+::::::: {.column width="50%"}
+
+- HapMap chr1 genetic map (250Mb)
+- Gutenkunst et al out-of-Africa model (3 pops)
+- mutation rate $2 \times 10^{-8}$ per gen
+- at $n=10^7$ 
+
+    * about 17 million variants
+    * VCF size: 318 TiB (250,000$\times$ larger)
+
+:::
+:::::::
+
+## Example: three samples; two trees; two variant sites
+
+![Example tree sequence](example_tree_sequence.png)
+
+-----------------------
+
+Storing a tree sequence in
+the four tables - *nodes*, *edges*, *sites*, and *mutations* -
+is *succinct* (no redundancy).
+
+. . .
+
+These are stored efficiently (hdf5) on disk
+with a bit more information (e.g., metadata).
+
+
+## Nodes and edges
+
+Edges 
+
+:   Common ancestor (i.e., coalescent) events.
+
+    Records: interval (left, right); parent node; child node.
+
+Nodes 
+
+:   The ancestors those happen in.
+
+    Records: time ago (of birth); ID (implicit).
+
+-------------------
+
+![Building a tree sequence](nodes_edges_walkthrough/nodes_edges_walkthrough.0.png)
+
+-------------------
+
+
+![Building a tree sequence](nodes_edges_walkthrough/nodes_edges_walkthrough.1.png)
+
+-------------------
+
+
+![Building a tree sequence](nodes_edges_walkthrough/nodes_edges_walkthrough.2.png)
+
+-------------------
+
+
+![Building a tree sequence](nodes_edges_walkthrough/nodes_edges_walkthrough.3.png)
+
+-------------------
+
+
+![Building a tree sequence](nodes_edges_walkthrough/nodes_edges_walkthrough.4.png)
+
+-------------------
+
+
+![Building a tree sequence](nodes_edges_walkthrough/nodes_edges_walkthrough.5.png)
+
+-------------------
+
+![Building a tree sequence](nodes_edges_walkthrough/nodes_edges_walkthrough.6.png)
+
+
+## Sites and mutations
+
+Mutations
+
+:   When state changes along the tree.
+
+    Records: site it occured at; derived state.
+
+Sites 
+
+:   Where mutations fall on the genome.
+
+    Records: genomic position; ancestral (root) state; ID (implicit).
+
+
+------------------
+
+![Adding mutations](sites_muts_walkthrough/sites_muts_walkthrough.0.png)
+
+------------------
+
+![Adding mutations](sites_muts_walkthrough/sites_muts_walkthrough.1.png)
+
+------------------
+
+![Adding mutations](sites_muts_walkthrough/sites_muts_walkthrough.2.png)
+
+------------------
+
+![Adding mutations](sites_muts_walkthrough/sites_muts_walkthrough.3.png)
+
+------------------
+
+![Adding mutations](sites_muts_walkthrough/sites_muts_walkthrough.4.png)
+
+
+
+<!-- 2. advertise tskit (1min) -->
```

Comment:
it's a bit packed. and when this is introduced folks don't yet have motivation to care about ts. migth save this till the end (like line 474)

(ok. well maybe they do with the graph you just showed of file sizes! but it seems a bit of a detour to me at this point in the talk)

Reply:
oops maybe this came off harsh as there is more than just the file sizes preceding this. so folks do have motivation to care as you've introduced them to both the moving between trees and file sizes bit. that said I stand by the 'detour' comment --- I think these notions about the broader utility of ts and the next steps would all fit better together at the end. 2c

Reply:
makes sense, i agree